lib/driver.js

```diff
@@ -225,12 +226,12 @@ class XCUITestDriver extends BaseDriver {
       device.setScaleFactor(this.opts.scaleFactor);
     }
 
+    await this.runReset();
+
     // handle logging
     this.logs = {};
     await this.startLogCapture();
 
-    await this.runReset();
```

Comment:
This was the initial idea - why should one perform reset at the end of the session? Usually it is very useful to keep the sate untouched, so you can verify what was happened during the test. Also, the next session will anyway perform reset on startup if needed.